Gems/GeoJSONSpawner/Code/Source/GeoJSONSpawner/ROS2Interface/GeoJSONSpawnerROS2Interface.cpp

```diff
@@ -0,0 +1,189 @@
+                        "The modify topic configuration.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &GeoJSONSpawnerROS2InterfaceConfiguration::m_deleteAllTopicConfiguration,
+                        "Delete All Topic Configuration",
+                        "The delete all enitities topic configuration.")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &GeoJSONSpawnerROS2InterfaceConfiguration::m_deleteByIdTopicConfiguration,
+                        "Delete By Id Topic Configuration",
+                        "The delete group of entities by id topic configuration.");
+            }
+        }
+    }
+
+    GeoJSONSpawnerROS2Interface::GeoJSONSpawnerROS2Interface(const GeoJSONSpawnerROS2InterfaceConfiguration& configuration)
+        : m_configuration(configuration)
+    {
+    }
+
+    void GeoJSONSpawnerROS2Interface::Reflect(AZ::ReflectContext* context)
+    {
+        GeoJSONSpawnerROS2InterfaceConfiguration::Reflect(context);
+
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<GeoJSONSpawnerROS2Interface, AZ::Component>()->Version(0)->Field(
+                "Configuration", &GeoJSONSpawnerROS2Interface::m_configuration);
+        }
+    }
+
+    void GeoJSONSpawnerROS2Interface::Activate()
+    {
+        m_spawnSubscriptionHandler = AZStd::make_unique<SubscriptionHandler<StringMsg>>(
+            [this](const StringMsg& message)
+            {
+                ProcessSpawnMessage(message);
+            });
+        m_spawnSubscriptionHandler->Activate(GetEntity(), m_configuration.m_spawnTopicConfiguration);
+
+        m_modifySubscriptionHandler = AZStd::make_unique<SubscriptionHandler<StringMsg>>(
+            [this](const StringMsg& message)
+            {
+                ProcessModifyMessage(message);
+            });
+        m_modifySubscriptionHandler->Activate(GetEntity(), m_configuration.m_modifyTopicConfiguration);
+
+        m_deleteAllSubscriptionHandler = AZStd::make_unique<SubscriptionHandler<EmptyMsg>>(
+            [this](const EmptyMsg& message)
+            {
+                ProcessDeleteAllMessage();
+            });
+        m_deleteAllSubscriptionHandler->Activate(GetEntity(), m_configuration.m_deleteAllTopicConfiguration);
+
+        m_deleteByIdSubscriptionHandler = AZStd::make_unique<SubscriptionHandler<StringMsg>>(
+            [this](const StringMsg& message)
+            {
+                ProcessDeleteByIdMessage(message);
+            });
+        m_deleteByIdSubscriptionHandler->Activate(GetEntity(), m_configuration.m_deleteByIdTopicConfiguration);
+
+        auto ros2Node = ROS2::ROS2Interface::Get()->GetNode();
+        if (ros2Node)
+        {
+            auto ros2Frame = ROS2::Utils::GetGameOrEditorComponent<ROS2::ROS2FrameComponent>(GetEntity());
```

Comment:
GetGameOrEditorComponent should not be used in this context.
It is used to get Game comments in GeneralWrapper in Editor context. It will fail with ROS2FrameComponent, since this component has separated Game and Editor components.
**And also use assert macro on everything that you assume to give expected result.**
It will make life easier if smth goes South, and reduce number of if statements.